miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/InstrumentRestController.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Instrument;
+import uk.ac.bbsrc.tgac.miso.core.data.type.InstrumentType;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginatedDataSource;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginationFilter;
+import uk.ac.bbsrc.tgac.miso.dto.DataTablesResponseDto;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.InstrumentDto;
+import uk.ac.bbsrc.tgac.miso.service.InstrumentService;
+
+@Controller
+@RequestMapping("/rest/instrument")
+public class InstrumentRestController extends RestController {
+  private final JQueryDataTableBackend<Instrument, InstrumentDto> jQueryBackend = new JQueryDataTableBackend<Instrument, InstrumentDto>() {
+    @Override
+    protected InstrumentDto asDto(Instrument model) {
+      return Dtos.asDto(model);
+    }
+
+    @Override
+    protected PaginatedDataSource<Instrument> getSource() throws IOException {
+      return instrumentService;
+    }
+  };
+
+  @Autowired
+  private InstrumentService instrumentService;
+
+  public void setLibraryService(InstrumentService instrumentService) {
+    this.instrumentService = instrumentService;
+  }
+
+  @RequestMapping(value = "/{instrumentId}", method = RequestMethod.GET, produces = "application/json")
+  @ResponseBody
+  public InstrumentDto getById(@PathVariable Long instrumentId) throws IOException {
+    Instrument r = instrumentService.get(instrumentId);
+    if (r == null) {
+      throw new RestException("No instrument found with ID: " + instrumentId, Status.NOT_FOUND);
+    }
+    return Dtos.asDto(r);
+  }
+
+  @RequestMapping(method = RequestMethod.GET, produces = "application/json")
+  @ResponseBody
+  public List<InstrumentDto> listAll() throws IOException {
+    return instrumentService.list().stream().map(Dtos::asDto).collect(Collectors.toList());
+  }
+
+  private static final Logger log = LoggerFactory.getLogger(InstrumentRestController.class);
+
+  @RequestMapping(method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.CREATED)
+  @ResponseBody
+  public InstrumentDto create(@RequestBody InstrumentDto instrumentDto, UriComponentsBuilder b, HttpServletResponse response)
+      throws IOException {
+    if (instrumentDto == null) {
+      throw new RestException("Cannot convert null to instrument", Status.BAD_REQUEST);
+    }
+    Long id = null;
+    try {
+      Instrument instrument = Dtos.to(instrumentDto);
+      id = instrumentService.create(instrument);
+    } catch (ConstraintViolationException | IllegalArgumentException e) {
+      log.error("Error while creating library. ", e);
+      RestException restException = new RestException(e.getMessage(), Status.BAD_REQUEST);
+      if (e instanceof ConstraintViolationException) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)